### DISCUSSION TITLE
Pipe Operator for NeLua

### DISCUSSION THREAD

**Discussion by ghost** [edubart/nelua-lang#253] [Ideas]
### PIPE OPERATOR

It would be very interesting if the language added the operator pipe (as in the functional languages R, F# and Elixir). With a **|>** to make reading and writing the code very readable and beautiful.

For example, for codes like:
`size(uppercase(string))`
Could it be:
`string |> uppercase |> size`

These is my idea for improving NeLua syntax. Thanks!

**Comment by asumbek** [comment]
In the main page of Nelua website
> If you know how to code in Lua then you probably know how to code in Nelua. It tries to have the same syntax, semantics, features, and APIs.

Nelua tries to be compliant with Lua 5.4, if there's a new major syntax in Nelua this will probably cause some issues with compliancy of Nelua.

You can just add your own pipe operator to your code, because Nelua preprocessor is powerful. You can simply modify the syntax grammar, add new AST parser, and then implement a new feature.
